File: app/src/main/java/ru/nehodov/weatherforecast/dao/CurrentLocationDao.java
package ru.nehodov.weatherforecast.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import ru.nehodov.weatherforecast.entities.CurrentLocation;

@Dao
public interface CurrentLocationDao {

    @Insert
    void insert(CurrentLocation location);

    @Query("SELECT * FROM currentlocation")
    LiveData<CurrentLocation> get();

    @Query("DELETE FROM currentlocation")
    void deleteAll();

    @Query("SELECT * FROM currentlocation")
    CurrentLocation getWithouLiveData();
}

File: app/src/main/java/ru/nehodov/weatherforecast/dao/CurrentLocationDao.kt
package ru.nehodov.weatherforecast.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import kotlinx.coroutines.flow.Flow
import ru.nehodov.weatherforecast.entities.CurrentLocation

@Dao
interface CurrentLocationDao {

    @Insert
    suspend fun insert(currentLocation: CurrentLocation)

    @Query("SELECT * FROM CurrentLocation")
    fun currentLocationData(): Flow<CurrentLocation>

    @Query("DELETE FROM CurrentLocation")
    suspend fun deleteAll()
}